Zend/zend_API.c

```diff
@@ -2019,7 +2019,7 @@ ZEND_API zend_module_entry* zend_register_module_ex(zend_module_entry *module) /
 	lcname = zend_string_alloc(name_len, 1);
 	zend_str_tolower_copy(ZSTR_VAL(lcname), module->name, name_len);
 
-	if ((module_ptr = zend_hash_add_mem(&module_registry, lcname, module, sizeof(zend_module_entry))) == NULL) {
+	if ((module_ptr = zend_hash_add_mem(&module_registry, zend_new_interned_string(lcname), module, sizeof(zend_module_entry))) == NULL) {
```

Comment:
This should be `lcname = zend_new_interned_string(lcname)` on a separate line. The semantics of zend_new_interned_string are similar to realloc. The argument is released, you just sometimes don't see it (if the string was not interned yet).